### PR TITLE
Jira Source - incremental sync

### LINKED ISSUE #1390: Support incremental sync in JIRA cloud
Follow up to #1389 

There are a few parts to this: 

1. Identify which streams can sync incrementally, and which bookmark you'll be using
1. Update the catalog generated in `discover` to declare which streams sync incrementally
1. Implement the incremental sync process

To incrementally sync a child entity child entity (e.g: `/customers/{customer_id}/customer_events` ), we will probably need to do a "full read" of the parent entity (get all parents), then look for the child entities that have changed since the last time we read the child entities. For example: if we're syncing `/customers/{customer_id}/customer_events`  then we should read all customers, then for each customer, read all the events we haven't read since the last time we performed a sync. It's probably not practical to keep track of the last event we synced for _each_  customer, so it may make sense to keep track of a global cursor for all customer_events. For example, we may just have to remember that the largest `customer_event.updated_at` value we extracted last time a sync ran was `october 10`. In which case the next sync we read all customers, then for each customer extract all the events where `updated_at >= october 10`.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368079337056) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Implementing incremental sync for Jira Source

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

Closes #1390

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py:23]: something is wrong here

File: airbyte-integrations/connectors/source-jira/source_jira/client.py [hunk 2]
```diff
@@ -142,10 +165,12 @@ def _get_custom_fields(self):
             if field.get("custom"):
                 yield field
 
-    def _get_issues_related(self, name):
+    def _get_issues_related(self, name, params={}):
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/source_jira/client.py:168]: mutable defaults

File: airbyte-integrations/connectors/source-jira/source_jira/entities.py [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/source_jira/entities.py:23]: same here

File: airbyte-integrations/connectors/source-jira/source_jira/entities.py [hunk 4]
```diff
@@ -33,7 +56,7 @@
     "issues": {
         "url": "/search",
         "extractor": lambda v: v["issues"],
-        "params": {**{"fields": ["attachment", "issuelinks", "security", "issuetype"]}},
+        "params": {**{"fields": ["attachment", "issuelinks", "security", "issuetype", "created"]}},
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/source_jira/entities.py:59]: `dict` nesting here seems unnecessary

**Review by keu — CHANGES_REQUESTED** [review]

**Review by keu — APPROVED** [review]

**Review by grebessi — APPROVED** [review]

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py:23]: this is expected, we changed the format for licenses

File: airbyte-integrations/connectors/source-jira/setup.py [hunk 5]
```diff
@@ -29,6 +29,6 @@
     author="Airbyte",
     author_email="contact@airbyte.io",
     packages=find_packages(),
-    install_requires=["airbyte-protocol", "base-python", "requests", "pytest==6.1.2"],
+    install_requires=["airbyte-protocol", "base-python", "requests", "pytest==6.1.2", "pendulum==2.1.2"],
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-jira/setup.py:32]: could you update it to use the pypi airbyte CDK package?

**Review by sherifnada — APPROVED** [review]
LGTM with a couple of comments:
1. could you add changelog.md
2. update to the pypi CDK

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py (hunk 1 above)
Comment (keu) [reply airbyte-integrations/connectors/source-jira/integration_tests/fixtures/create_projects.py:23]: my comment was about two license headers in one file

**Pull request merged**